### PR TITLE
Site : add user site.

### PR DESCRIPTION
**Pull request by RamBoFe**

### CODE REVIEW

File: api/campuses.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export const ENTITY_PLURAL = 'campuses';
+
+export default axios => ({
+  async getCampuses(mask, { search = null } = {}) {
+    const response = await axios.get(
+      `/${ENTITY_PLURAL}/infos`,
```
Comment (Nainterceptor) [inline api/campuses.js]: As said in API, is it possible to use get api endpoint ?
And, it's not scoped to campus here

File: api/campuses.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+export const ENTITY_PLURAL = 'campuses';
+
+export default axios => ({
+  async getCampuses(mask, { search = null } = {}) {
```
Comment (Nainterceptor) [inline api/campuses.js]: search param seems to be useless

File: api/forms.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+export const ENTITY = 'form';
+
+export default axios => ({
+  async postFormContact(fields) {
+    return axios.post(
+      `/${ENTITY_PLURAL}/contact`,
+      qs.stringify(fields),
+      {
+        headers: {
+          'Content-Type': 'application/x-www-form-urlencoded',
```
Comment (Nainterceptor) [inline api/forms.js]: why not JSON ? It's JSON everywhere in the app

File: assets/css/elements/button.scss [hunk 4]
```diff
@@ -1,4 +1,6 @@
+/*
```
Comment (Nainterceptor) [inline assets/css/elements/button.scss]: If the style become useless, please remove it, but check the ride view before

File: assets/css/elements/form.scss [hunk 5]
```diff
@@ -1,3 +1,4 @@
+/*
```
Comment (Nainterceptor) [inline assets/css/elements/form.scss]: If the style become useless, please remove it, but check the ride view before

File: assets/css/main.scss [hunk 6]
```diff
@@ -30,17 +31,17 @@
...
+@import "~bulma/sass/grid/columns.sass";
+@import "~bulma/sass/grid/tiles.sass";
 
 //@import "~bulma/sass/layout/hero.sass";
-//@import "~bulma/sass/layout/section.sass";
+@import "~bulma/sass/layout/section.sass";
 
 
-html, body, #__nuxt, #__layout {
+/*html, body, #__nuxt, #__layout {
```
Comment (Nainterceptor) [inline assets/css/main.scss]: If the style become useless, please remove it, but check the ride view before

File: components/footer.vue [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+      <div class="column is-3 has-text-centered-mobile">
+        <img
+          src="logo_ec_blue.svg"
+          alt="E-chauffeur"
+        >
+      </div>
+      <div class="column has-text-centered-mobile">
+        <img
+          src="logo_dirisi.png"
+          alt="Dirisi"
```
Comment (Nainterceptor) [inline components/footer.vue]: Is it Dirisi or DIRISI ?

File: components/footer.vue [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+        <img
+          src="logo_fab_num.png"
+          alt="Fabrique Numérique"
+        >
+      </div>
+    </div>
+  </footer>
+</template>
+
+<script>
```
Comment (Nainterceptor) [inline components/footer.vue]: This part seems to be useless

File: components/header.vue [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+          </nuxt-link>
+        </div>
+      </div>
+    </nav>
+  </header>
+</template>
+
+<script>
+export default {
+  name: 'Header',
```
Comment (Nainterceptor) [inline components/header.vue]: name is probably useless here

File: components/header.vue [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+        <div class="navbar-end">
+          <nuxt-link
+            class="navbar-item"
+            to="/"
+          >
+            Accueil
+          </nuxt-link>
+          <nuxt-link
+            class="navbar-item"
+            to="/contact"
```
Comment (Nainterceptor) [inline components/header.vue]: It's better to use verbose `:to="{name: 'contact'}"`, so when we will change the URL, we will not have to change links

File: components/header.vue [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+        class="navbar-menu"
+        :class="{
+          'is-active': isBurgerMenu,
+          'is-blue': isBurgerMenu && isBurgerMenuBlueColor
+        }"
+      >
+        <div class="navbar-end">
+          <nuxt-link
+            class="navbar-item"
+            to="/"
```
Comment (Nainterceptor) [inline components/header.vue]: It's better to use verbose `:to="{name: 'contact'}"`, so when we will change the URL, we will not have to change links (to add i18n, for the index case)

File: components/header.vue [hunk 12]
```diff
@@ -0,0 +1,125 @@
+<template>
+  <header>
+    <nav class="navbar">
+      <div class="navbar-brand">
+        <nuxt-link
+          class="navbar-item"
+          to="/"
```
Comment (Nainterceptor) [inline components/header.vue]: It's better to use verbose `:to="{name: 'contact'}"`, so when we will change the URL, we will not have to change links (to add i18n, for the index case)

File: layouts/default.vue [hunk 13]
```diff
@@ -1,3 +1,122 @@
 <template>
-  <nuxt />
+  <div class="wrapper">
```
Comment (Nainterceptor) [inline layouts/default.vue]: Have you checked the ride view ? Please do it, I think that will become broken with the layout edit

File: components/header.vue [hunk 14]
```diff
@@ -0,0 +1,125 @@
...
+  data() {
+    return {
+      isBurgerMenu: null,
+      isBurgerMenuBlueColor: null,
+    };
+  },
+
+  watch: {
+    '$route.path': function () {
+      this.init();
```
Comment (Nainterceptor) [inline components/header.vue]: Maybe an anti-pattern, you should not have to do that, you probably just have to pass a parameter to the component

File: layouts/default.vue [hunk 15]
```diff
@@ -1,3 +1,122 @@
...
+    return {
+      showShape: {
+        index: null,
+        contact: null,
+      },
+      logoPath: null,
+    };
+  },
+
+  watch: {
```
Comment (Nainterceptor) [inline layouts/default.vue]: Maybe an anti-pattern, you should not have to do that, you probably just have to pass a parameter to the component

File: nuxt.config.js [hunk 16]
```diff
@@ -1,17 +1,17 @@
...
 
   /*
   ** Headers of the page
   */
   head: {
-    title: 'e-Chauffeur',
-    htmlAttrs: {
-      lang: 'fr',
-    },
+    title: pkg.name,
```
Comment (Nainterceptor) [inline nuxt.config.js]: Why ?

File: nuxt.config.js [hunk 17]
```diff
@@ -1,17 +1,17 @@
...
     meta: [
       { charset: 'utf-8' },
       { name: 'viewport', content: 'width=device-width, initial-scale=1' },
+      { hid: 'description', name: 'description', content: pkg.description },
```
Comment (Nainterceptor) [inline nuxt.config.js]: Why ?

File: nuxt.config.js [hunk 18]
```diff
@@ -72,11 +73,4 @@ module.exports = {
     position: 'bottom-right',
     duration: 15000,
   },
-
```
Comment (Nainterceptor) [inline nuxt.config.js]: Seems to be an error on merging

File: nuxt.config.js [hunk 19]
```diff
@@ -50,7 +52,6 @@ module.exports = {
     'nuxt-leaflet',
     '@nuxtjs/axios',
     '@nuxtjs/toast',
-    '@nuxtjs/pwa',
```
Comment (Nainterceptor) [inline nuxt.config.js]: Seems to be a fail on merging

File: package.json [hunk 20]
```diff
@@ -13,34 +13,35 @@
     "precommit": "npm run lint"
   },
   "dependencies": {
```
Comment (Nainterceptor) [inline package.json]: Why downgrades ?

File: pages/contact.vue [hunk 21]
```diff
@@ -0,0 +1,387 @@
...
+        </div>
+      </div>
+    </section>
+
+    <form @submit.prevent="sendForm">
+      <div class="columns">
+        <div class="column">
+          <ec-field
+            :id="'ec-form-prenom'"
+            :label="'votre prénom'"
```
Comment (Nainterceptor) [inline pages/contact.vue]: You should be able to write `label="votre prénom"`

File: pages/contact.vue [hunk 22]
```diff
@@ -0,0 +1,387 @@
...
+          </p>
+        </div>
+      </div>
+    </section>
+
+    <form @submit.prevent="sendForm">
+      <div class="columns">
+        <div class="column">
+          <ec-field
+            :id="'ec-form-prenom'"
```
Comment (Nainterceptor) [inline pages/contact.vue]: You should be able to write `id="ec-form-prenom"` And, in code, please use english word

File: pages/contact.vue [hunk 23]
```diff
@@ -0,0 +1,387 @@
...
+            :label="'votre prénom'"
+          >
+            <input
+              id="ec-form-prenom"
+              v-model="fields.firstname"
+              v-validate="'required'"
```
Comment (Nainterceptor) [inline pages/contact.vue]: required="true" is not enough ?

File: pages/contact.vue [hunk 24]
```diff
@@ -0,0 +1,387 @@
...
+              type="text"
+              class="input"
+              name="firstname"
+              data-vv-as="prénom"
+              :class="{ 'is-danger': errors.has('firstname') }"
+              :placeholder="firstLetterCapitalize('tapez votre prénom')"
```
Comment (Nainterceptor) [inline pages/contact.vue]: Ok, using this helper may be useful for user input, but why here ?

File: pages/contact.vue [hunk 25]
```diff
@@ -0,0 +1,387 @@
...
+            <input
+              id="ec-form-nom"
+              v-model="fields.lastname"
+              v-validate="'required'"
+              type="text"
+              class="input"
+              name="lastname"
+              data-vv-as="nom"
+              :class="{ 'is-danger': errors.has('lastname') }"
+              :placeholder="firstLetterCapitalize('tapez votre nom')"
```
Comment (Nainterceptor) [inline pages/contact.vue]: Ok, using this helper may be useful for user input, but why here ?

File: pages/contact.vue [hunk 26]
```diff
@@ -0,0 +1,387 @@
...
+            <input
+              id="ec-form-email"
+              v-model="fields.email"
+              v-validate="'required|email'"
+              type="email"
+              class="input"
+              name="email"
+              data-vv-as="courriel"
+              :class="{ 'is-danger': errors.has('email') }"
+              :placeholder="firstLetterCapitalize('tapez votre courriel')"
```
Comment (Nainterceptor) [inline pages/contact.vue]: Ok, using this helper may be useful for user input, but why here ?

File: pages/contact.vue [hunk 27]
```diff
@@ -0,0 +1,387 @@
...
+            <input
+              id="ec-form-phone"
+              v-model="fields.phone"
+              v-validate="'numeric'"
+              type="tel"
+              class="input"
+              name="phone"
+              data-vv-as="téléphone"
+              :class="{ 'is-danger': errors.has('phone') }"
+              :placeholder="firstLetterCapitalize('tapez votre numéro de téléphone')"
```
Comment (Nainterceptor) [inline pages/contact.vue]: Ok, using this helper may be useful for user input, but why here ?

File: pages/contact.vue [hunk 28]
```diff
@@ -0,0 +1,387 @@
...
+
+  computed: {
+    isBtnSubDisabled() {
+      return (!this.fields.firstname || !this.fields.lastname || !this.fields.gsbdd
+          || !this.fields.email || !this.fields.message);
+    },
+  },
+
+  methods: {
+    firstLetterCapitalize(text) {
```
Comment (Nainterceptor) [inline pages/contact.vue]: Finally, seems to be useless method

File: pages/contact.vue [hunk 29]
```diff
@@ -0,0 +1,387 @@
...
+            >
+              <select
+                id="ec-form-gsbdd"
+                v-model="fields.gsbdd"
+                v-validate="'required'"
+                name="gsbdd"
+                :class="{ 'placeholder': !fields.gsbdd }"
+              >
+                <option value="">
+                  {{ firstLetterCapitalize('sélectionnez votre GSBdD') }}
```
Comment (Nainterceptor) [inline pages/contact.vue]: Ok, using this helper may be useful for user input, but why here ?

File: pages/contact.vue [hunk 30]
```diff
@@ -0,0 +1,387 @@
...
+            :label="'votre message'"
+          >
+            <textarea
+              id="ec-form-message"
+              v-model="fields.message"
+              v-validate="'required'"
+              class="textarea"
+              name="message"
+              :class="{ 'is-danger': errors.has('message') }"
+              :placeholder="firstLetterCapitalize('tapez votre message')"
```
Comment (Nainterceptor) [inline pages/contact.vue]: Ok, using this helper may be useful for user input, but why here ?

File: pages/contact.vue [hunk 31]
```diff
@@ -0,0 +1,387 @@
...
+      <div class="columns">
+        <div class="column">
+          <div class="control has-text-right">
+            <button
+              class="button is-primary submit"
+              :class="{ 'is-loading': pending }"
+              :disabled="isBtnSubDisabled"
+            >
+              <span class="message">
+                <em>envoyer</em> mon <em>message</em>
```
Comment (Nainterceptor) [inline pages/contact.vue]: Capitalize the first letter

File: pages/contact.vue (hunk 22 above)
Comment (Nainterceptor) [reply pages/contact.vue]: Please consider this change in other fields

File: pages/contact.vue (hunk 21 above)
Comment (Nainterceptor) [reply pages/contact.vue]: Please consider this change in other fields

File: pages/contact.vue [hunk 32]
```diff
@@ -0,0 +1,387 @@
...
+      </button>
+    </div>
+  </main>
+</template>
+
+<script>
+import ecField from '~/components/form/field.vue';
+
+export default {
+  name: 'Contact',
```
Comment (Nainterceptor) [inline pages/contact.vue]: name is useless in our method of work with vue

File: pages/index.vue [hunk 33]
```diff
@@ -1,57 +1,205 @@
...
+          {{ campuse.id }}
+        </a>
+      </div>
+
+      <div class="columns">
+        <div class="column is-4 has-text-centered-mobile bn-infos">
+          <h1 class="title">
+            Informations
+          </h1>
+          <vue-simple-markdown :source="informations" />
```
Comment (Nainterceptor) [inline pages/index.vue]: informations -> information

Comment (Nainterceptor) [reply pages/index.vue]: (For the whole file, except in french strings)

File: plugins/validator.js [hunk 34]
```diff
@@ -0,0 +1,21 @@
+import Vue from 'vue';
+import VeeValidate from 'vee-validate';
+
+const dictionary = {
+  fr: {
+    messages: {
+      required: field => `Le champ ${field} est recquis.`,
```
Comment (Nainterceptor) [inline plugins/validator.js]: typo here "requis"

File: plugins/validator.js [hunk 35]
```diff
@@ -0,0 +1,21 @@
...
+      email: field => `Le champ ${field} doit être une addresse courriel valide.`,
```
Comment (Nainterceptor) [inline plugins/validator.js]: type here "adresse"

File: plugins/validator.js [hunk 36]
```diff
@@ -0,0 +1,21 @@
...
+      numeric: field => `Le champ ${field} ne doit contenir que des caractères numériques.`,
```
Comment (Nainterceptor) [inline plugins/validator.js]: maybe it's better to say "chiffres"

File: api/campuses.js (hunk 1 above)
Comment (RamBoFe) [reply api/campuses.js]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: api/campuses.js (hunk 2 above)
Comment (RamBoFe) [reply api/campuses.js]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: api/forms.js (hunk 3 above)
Comment (RamBoFe) [reply api/forms.js]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: assets/css/elements/form.scss (hunk 5 above)
Comment (RamBoFe) [reply assets/css/elements/form.scss]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: assets/css/elements/button.scss (hunk 4 above)
Comment (RamBoFe) [reply assets/css/elements/button.scss]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: assets/css/main.scss (hunk 6 above)
Comment (RamBoFe) [reply assets/css/main.scss]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: components/footer.vue (hunk 7 above)
Comment (RamBoFe) [reply components/footer.vue]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: components/footer.vue (hunk 8 above)
Comment (RamBoFe) [reply components/footer.vue]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: components/header.vue (hunk 9 above)
Comment (RamBoFe) [reply components/header.vue]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: components/header.vue (hunk 10 above)
Comment (RamBoFe) [reply components/header.vue]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: components/header.vue (hunk 11 above)
Comment (RamBoFe) [reply components/header.vue]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: components/header.vue (hunk 12 above)
Comment (RamBoFe) [reply components/header.vue]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: layouts/default.vue (hunk 13 above)
Comment (RamBoFe) [reply layouts/default.vue]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: components/header.vue (hunk 14 above)
Comment (RamBoFe) [reply components/header.vue]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: layouts/default.vue (hunk 15 above)
Comment (RamBoFe) [reply layouts/default.vue]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: nuxt.config.js (hunk 16 above)
Comment (RamBoFe) [reply nuxt.config.js]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: nuxt.config.js (hunk 17 above)
Comment (RamBoFe) [reply nuxt.config.js]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: nuxt.config.js (hunk 18 above)
Comment (RamBoFe) [reply nuxt.config.js]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: nuxt.config.js (hunk 19 above)
Comment (RamBoFe) [reply nuxt.config.js]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: package.json (hunk 20 above)
Comment (RamBoFe) [reply package.json]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: pages/contact.vue (hunk 21 above)
Comment (RamBoFe) [reply pages/contact.vue]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: pages/contact.vue (hunk 22 above)
Comment (RamBoFe) [reply pages/contact.vue]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: pages/contact.vue (hunk 23 above)
Comment (RamBoFe) [reply pages/contact.vue]: I use the package vee validate for the form validator that's why i don't use required=true.

File: pages/contact.vue (hunk 24 above)
Comment (RamBoFe) [reply pages/contact.vue]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: pages/contact.vue (hunk 25 above)
Comment (RamBoFe) [reply pages/contact.vue]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: pages/contact.vue (hunk 26 above)
Comment (RamBoFe) [reply pages/contact.vue]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: pages/contact.vue (hunk 27 above)
Comment (RamBoFe) [reply pages/contact.vue]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: pages/contact.vue (hunk 28 above)
Comment (RamBoFe) [reply pages/contact.vue]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: pages/contact.vue (hunk 29 above)
Comment (RamBoFe) [reply pages/contact.vue]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: pages/contact.vue (hunk 30 above)
Comment (RamBoFe) [reply pages/contact.vue]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: pages/contact.vue (hunk 31 above)
Comment (RamBoFe) [reply pages/contact.vue]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: pages/contact.vue (hunk 32 above)
Comment (RamBoFe) [reply pages/contact.vue]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: pages/index.vue (hunk 33 above)
Comment (RamBoFe) [reply pages/index.vue]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: plugins/validator.js (hunk 34 above)
Comment (RamBoFe) [reply plugins/validator.js]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: plugins/validator.js (hunk 35 above)
Comment (RamBoFe) [reply plugins/validator.js]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: plugins/validator.js (hunk 36 above)
Comment (RamBoFe) [reply plugins/validator.js]: Fix on 1e9aec86fde515fef3f8862633c4195b0cc0ea1a

File: pages/index.vue [hunk 37]
```diff
@@ -1,57 +1,238 @@
...
+      campuses: [],
+      information: '',
+      image: '',
+      btnActive: {},
+    };
+  },
+
+  async created() {
+    this.campuses = await this.getCampuses();
+    this.setInformation(this.campuses[0].id);
```
Comment (Nainterceptor) [inline pages/index.vue]: What if campuses array contain less than 1 element ?

File: api/forms.js [hunk 38]
```diff
@@ -0,0 +1,16 @@
+const CONTENT_TYPE_JSON = 'application/json';
+export const ENTITY_PLURAL = 'forms';
+
+export default axios => ({
+  async postFormContact(fields) {
+    return axios.post(
+      `/${ENTITY_PLURAL}/contact`,
+      JSON.stringify(fields),
```
[the remark this answers is not in the document]
Comment (RamBoFe) [reply api/forms.js]: Fix on ec931056525594dde88bb78b3486b0179eab5ae1

File: api/forms.js [hunk 39]
```diff
@@ -0,0 +1,16 @@
...
+      {
+        headers: {
+          'Content-Type': CONTENT_TYPE_JSON,
```
[the remark this answers is not in the document]
Comment (RamBoFe) [reply api/forms.js]: Fix one c931056525594dde88bb78b3486b0179eab5ae1

File: pages/index.vue (hunk 37 above)
Comment (RamBoFe) [reply pages/index.vue]: Fix on ec931056525594dde88bb78b3486b0179eab5ae1

**Pull request merged**